### ISSUE TITLE
Does not get vo or hwdec settings from ~/.config/mpv.conf with recent libmpv1

### ISSUE THREAD

**Issue by mc4man** [u8sand/Baka-MPlayer#167]
This is seen here with libmpv1 from mpv  0.11, 0.12, git master.  So baka just uses a vo of opengl
With libmpv1 from mpv 0.10 baka does use the vo & hwdec specified in ~/.config/mpv.conf

A fresh bakamplayer.ini from 0.11 > git master shows this - 
   "mpv": {
        "screenshot-directory": ".",
        "screenshot-format": "",
        "screenshot-template": "screenshot%#04n",
        "speed": 1,
        "vo": "",
        "volume": 100
    },

I have to manually edit to specify vo & hwdec, ie.
    "mpv": {
        "hwdec": "auto",
        "screenshot-directory": ".",
        "screenshot-format": "",
        "screenshot-template": "screenshot%#04n",
        "speed": 1,
        "vo": "opengl-hq",
        "volume": 100
    },

**Comment by u8sand** [comment]
@mc4man we've never got settings from mpv's config file, this is largely intended. What, specifically, would you be interested in changing?

**Comment by mc4man** [comment]
That's wierd then - on libmpv from 0.10 if I make a change in mpv.conf concerning vo or hwdec it's reflected in baka. For instance if I set vo=vdpau then baka shows vo of vdpau, if I set to opengl-hq then baka shows opengl-hq. (in the view >

**Comment by mc4man** [comment]
Sorry, went to ck. on something before finishing comment above & it posted anyway?
Meant to say in view > Show playback info (tab)
So am a bit confused as how to enable hwdec or change the vo from opengl if not thru ~/.config/mpv.conf or manually editing the .ini.

**Comment by u8sand** [comment]
This is related to #172 --it has to do with mpv/conf being used at all. Both will most likely be fixed at the same time.

**Comment by mc4man** [comment]
Using reasonably current all, (ubuntu 16.04, baka 2.0.4, libmpv 0.14),  I can say that baka doesn't use mpv.conf nor is it affected at all by mpv.conf in any way I can tell which is ok, many options/profiles/protocols in my mpv.conf would be of no use to baka anyway.
baka does do a good job with hwdec (hwdec=auto), actually a bit more efficient than mpv in many cases. So if it's ok to add hwdec=something & vo=something to bakamplayer.ini manually then why not include such options in baka-mplayer > settings > preferences?

**Comment by u8sand** [comment]
@mc4man @godly-devotion would it make sense to have a tab for mpv settings and just have a textbox where you can put mpv settings in mpv's format in the preferences dialog?

**Comment by godly-devotion** [comment]
Yeah I think that would be a good idea. It could be displayed as a table just like Key Bindings (key, value pair).

**Comment by mc4man** [comment]
That could work, I'd probably like to see/test what would happen if 'wrong' combo of hwdec & gpu or hwdec & vo were user picked.
If possible to note for hwdec that the preferred/recommended setting  would be "auto"  that would also be good & eliminate any issues of a user mismatched hwdec selection & gpu in use.

**Comment by u8sand** [comment]
@godly-devotion my main motivation for just a textbox, is plenty of users may already have perfectly good mpv configurations, they should be able to copy and paste those in baka.

**Comment by godly-devotion** [comment]
@u8sand You bring up a good point where users may already have a mpv config file they might want to use. What if we just respected that config file (maybe allow them to specify the location of the config file if they placed it elsewhere under Pref) but override only the values that need to be from Baka (which is already being done)?

**Comment by u8sand** [comment]
@godly-devotion My issue with that is, the mpv config file is inherently different from baka's its keybindings are quite different, handling all of those would most likely be annoying. 

We could change baka's settings file again to be exactly like mpv's configuration but with more options, then we load what is for baka and discard the rest.

We could also just have an config import function that ports an existing mpv configuration to baka.

**Comment by LoathingKernel** [comment]
In my opinion you should not load the configuration for mpv simply because the player part of mpv and baka are not the same program, their only similarity is that they are both using libmpv as a back-end. So, in my mind, different programs = diffenent configurations. For example, SMPlayer doesn't honor mplayer.conf.

I do not think a textbox would be a good idea, it makes editing the configuration annoying and it looks bad. The best option in my opinion would be to add the ability to import the `mpv.conf` if the user wants in the form of a button and load the extra configurations options that are not already controlled by an existing setting into a table similar to the keybinding.

As for the configuration format, I believe you are on the right path right now with json, as it is handled gracefully by both mpv ipc and Qt.

**Comment by u8sand** [comment]
@LoathingKernel JSON's tree structure has another benefit; consider this gist: https://gist.github.com/u8sand/dd8623c13c242b2e0697 

It may be possible with QMetaProperties to define all save-able object properties with Q_PROPERTY and enumerate through them for loading/saving.

I like the idea of importing some of mpv.conf and think we should try to put anything configurable in the Preferences Dialog.

**Comment by LoathingKernel** [comment]
@u8sand Yes, I see what you mean and completely agree, this simplifies the way settings are handled, and enables you to add new properties in an easy way.

Putting anything configurable in the settings might be a huge task, mpv configuration options are numerous, confusing and, many of them, obscure. Baka tries to be simple and enjoyable and adding numerous knobs and switches is a step away from that. What if the users were asked, when they load their `mpv.conf`, to upload it automatically somewhere, and use that information to only implement the more common options? mpv configuration has hardly any sensitive information for the user to keep it a secret. If a user is especially particular and wants a specific option, just keep that table around under the advanced tab and let him play with mpv options until he breaks it.

**Comment by godly-devotion** [comment]
So to summarize @u8sand and @LoathingKernel, we will add ability under Preferences to select a mpv config file you'd like to use. Great part is this is relatively easy to do.

**Comment by darkbasic** [comment]
I need af=drc=2:0.75 to be able to actually hear DTS tracks with my laptop's built-in speakers, any chance to tell baka to read my mpv.conf file?
